imports/client/components/PuzzleListPage.tsx

```diff
@@ -305,33 +305,18 @@ class PuzzleListView extends React.Component<PuzzleListViewProps, PuzzleListView
       case 'group': {
         const puzzleGroups = this.puzzleGroupsByRelevance();
         const groupComponents = puzzleGroups.map((g) => {
-          if (g.sharedTag) {
-            return (
-              <RelatedPuzzleGroup
-                key={g.sharedTag._id}
-                sharedTag={g.sharedTag}
-                relatedPuzzles={g.puzzles}
-                allTags={this.props.allTags}
-                includeCount={false}
-                layout="grid"
-                canUpdate={this.props.canUpdate}
-              />
-            );
-          } else {
-            return (
-              <div key="ungrouped" className="puzzle-list-ungrouped">
-                <div>Puzzles in no group:</div>
-                <div className="puzzles">
-                  <PuzzleList
-                    puzzles={g.puzzles}
-                    allTags={this.props.allTags}
-                    layout="grid"
-                    canUpdate={this.props.canUpdate}
-                  />
-                </div>
-              </div>
-            );
-          }
+          return (
+            <RelatedPuzzleGroup
+              key={g.sharedTag ? g.sharedTag._id : 'ungrouped'}
+              sharedTag={g.sharedTag || null}
```

Comment:
Prefer `undefined` over `null` in general, unless there's a concrete reason you have to use `null` (like compatibility with a particular API).

```suggestion
              sharedTag={g.sharedTag}
```

Reply:
I used null just because it’s harder to do by accident, so type checking offers a little more benefit, but I’m happy to change it.